contents/julia_101.md

```diff
@@ -423,55 +517,2108 @@ logarithm(10)
+"""
+my_matrix = Matrix{Float64}(undef, 10, 2)
+"""
+)
+```
+
+We also have some **syntax aliases** for the most common elements in array construction:
+
+* `zeros` for all elements being initialized to value zero.
+  Note that the default type is `Float64` which can be changed if necessary:
+
+    ```jl
+    sco(
+    """
+    my_vector_zeros = zeros(10)
+    """
+    )
+    ```
+
+    ```jl
+    sco(
+    """
+    my_matrix_zeros = zeros(Int64, 10, 2)
+    """
+    )
+    ```
+
+* `ones` for all elements being initialized to value one:
+
+    ```jl
+    sco(
+    """
+    my_vector_ones = ones(Int64, 10)
+    """
+    )
+    ```
+
+    ```jl
+    sco(
+    """
+    my_matrix_ones = ones(10, 2)
+    """
+    )
+    ```
+
+For other elements we can first intantiate an array with `undef` elements and use the `fill!` function to fill all elements of an array with the desired element.
+Here's an example with `3.14` ($\pi$):
+
+```jl
+sco(
+"""
+my_matrix_π = Matrix{Float64}(undef, 2, 2)
+fill!(my_matrix_π, 3.14)
+"""
+)
+```
+
+We can also create arrays with **arrays literals**.
+For example a 2x2 matrix of integers:
+
+```jl
+sco(
+"""
+[[1 2]
+ [3 4]]
+"""
+)
+```
+
+Array literals also accept a type specification before the `[]` brackets.
+So, if we want the same 2x2 array as before but now as floats, we can do so:
+
+```jl
+sco(
+"""
+Float64[[1 2]
+        [3 4]]
+"""
+)
+```
+
+It also works for vectors:
+
+```jl
+sco(
+"""
+Bool[0, 1, 0, 1]
+"""
+)
+```
+
+You can even **mixmatch** array literals with the constructors:
+
+```jl
+sco(
+"""
+[ones(Int, 2, 2) zeros(Int, 2, 2)]
+"""
+)
+```
+
+```jl
+sco(
+"""
+[zeros(Int, 2, 2)
+ ones(Int, 2, 2)]
+"""
+)
+```
+
+
+```jl
+sco(
+"""
+[ones(Int, 2, 2) [1; 2]
+ [3 4]            5]
+"""
+)
+```
+
+Another powerful way to create arrays are **array comprehensions**.
+You specify what you want to do inside the `[]` brackets.
+For example, say we want to create a vector of squares from 1 to 100:
+
+```jl
+sco(
+"""
+[x^2 for x ∈ 1:10]
+"""
+)
+```
+
+They also support multiple inputs:
+
+```jl
+sco(
+"""
+[x*y for x ∈ 1:10 for y ∈ 1:2]
+"""
+)
+```
+
+And conditionals:
+
+```jl
+sco(
+"""
+[x^2 for x ∈ 1:10 if isodd(x)]
+"""
+)
+```
+
+As with array literals you can specify your desired type before the `[]` brackets:
+
+
+```jl
+sco(
+"""
+Float64[x^2 for x ∈ 1:10 if isodd(x)]
+"""
+)
+```
+
+Finally, we can also create arrays with **concatenation functions**:
+
+* `cat`: concatenate input arrays along a specific dimension `dims`
+
+    ```jl
+    sco(
+    """
+    cat(ones(2), zeros(2), dims=1)
+    """
+    )
+    ```
+
+    ```jl
+    sco(
+    """
+    cat(ones(2), zeros(2), dims=2)
+    """
+    )
+    ```
+
+* `vcat`: vertical concatenation, a shorthand for `cat(...; dims=1)`
+
+    ```jl
+    sco(
+    """
+    vcat(ones(2), zeros(2))
+    """
+    )
+    ```
+
+* `hcat`: horizontal concatenation, a shorthand for `cat(...; dims=2)`
+
+    ```jl
+    sco(
+    """
+    hcat(ones(2), zeros(2))
```

Comment:
and maybe add a mention for `[a; b]`, because that is suggested [here](https://docs.julialang.org/en/v1/manual/style-guide/#Don't-overuse-...)

Reply:
Too much complexity... Let's stick with hcat for now